### PR TITLE
Bean for logged gateway discord client

### LINKED ISSUE #20: Instantiate a Bean for a logged-in GatewayDiscordClient
- The `Bean` method should return a logged-in `GatewayDiscordClient`
  - The token should be supplied in `application.properties` and/or an environment variable
  - No event handlers are needed at this point
- The `Bean` method should be in a `Configuration` class called `GatewayDiscordClientConfig` or something similar
- The `Configuration` class should be in the package `com.mitchelltford.discordbot` or an appropriately-named sub-package
- The `Bean` method should be called `gatewayDiscordConfig` or something similar

### PR DESCRIPTION
**Pull request by LiangZhang21**
closes #20 

-Was able to set up the bean and build the discord bot.
-Was able to have the bot log into discord.

commits collaborated with @nickvichai

### CODE REVIEW

File: src/main/java/com/mitchelltford/discordbot/GatewayDiscordClientConfig.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Bean;
+import discord4j.core.GatewayDiscordClient;
+import discord4j.core.DiscordClientBuilder;
+
+
+@Configuration
+public class GatewayDiscordClientConfig {
+
+  private GatewayDiscordClient client;
```
Comment (MitchellTFord) [inline src/main/java/com/mitchelltford/discordbot/GatewayDiscordClientConfig.java:12]:
```suggestion
```
This class shouldn't need any instance variables. It doesn't look like this was being used anyway

File: src/main/java/com/mitchelltford/discordbot/Application.java [hunk 2]
```diff
@@ -2,12 +2,17 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ConfigurableApplicationContext;
 
 @SpringBootApplication
 public class Application {
```
Comment (MitchellTFord) [inline src/main/java/com/mitchelltford/discordbot/Application.java:9]: All of the changes in `Application.java` are unnecessary and can be reverted

File: src/main/java/com/mitchelltford/discordbot/GatewayDiscordClientConfig.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  @Bean
+  public GatewayDiscordClient gatewayDiscordConfig() {
+//    client = DiscordClientBuilder.create("***Token Goes Here***").build().login().block();
```
Comment (MitchellTFord) [inline src/main/java/com/mitchelltford/discordbot/GatewayDiscordClientConfig.java:16]: The authorization token should come from a properties file or an environment variable (or kind of both, depending on the implementation)

**Review by MitchellTFord — CHANGES_REQUESTED** [review]
A good start, lads (minus the whole leaking of secrets thing)

File: src/main/java/com/mitchelltford/discordbot/GatewayDiscordClientConfig.java (hunk 1 above)
Comment (nickvichai) [reply src/main/java/com/mitchelltford/discordbot/GatewayDiscordClientConfig.java:12]: It's being used in the Bean Method.

File: src/main/java/com/mitchelltford/discordbot/Application.java (hunk 2 above)
Comment (nickvichai) [reply src/main/java/com/mitchelltford/discordbot/Application.java:9]: Sounds good, will remove ApplicationContext, but we need to use the ConfigurableApplicationContext.

Comment (MitchellTFord) [reply src/main/java/com/mitchelltford/discordbot/Application.java:9]: nah, I meant **all** the changes in the file, not just the one GitHub is previewing

File: src/main/java/com/mitchelltford/discordbot/GatewayDiscordClientConfig.java (hunk 1 above)
Comment (MitchellTFord) [reply src/main/java/com/mitchelltford/discordbot/GatewayDiscordClientConfig.java:12]: You don't need to handle the singleton creation and storage, Spring does all of that for you. You just need to write the Bean method

File: src/main/java/com/mitchelltford/discordbot/Application.java [hunk 4]
```diff
@@ -1,18 +1,13 @@
 package com.mitchelltford.discordbot;
 
+import java.io.IOException;
```
Comment (MitchellTFord) [inline src/main/java/com/mitchelltford/discordbot/Application.java:3]:
```suggestion
import java.io.IOException;
```
Removed unused import

**Review by MitchellTFord — CHANGES_REQUESTED** [review]

File: src/main/java/com/mitchelltford/discordbot/GatewayDiscordClientConfig.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.context.annotation.Bean;
+import discord4j.core.GatewayDiscordClient;
+import discord4j.core.DiscordClientBuilder;
+
+
+@Configuration
+public class GatewayDiscordClientConfig {
+
+  @Bean
+  public GatewayDiscordClient gatewayDiscordConfig(@Value("${bot.token}") String token) {
```
Comment (MitchellTFord) [inline src/main/java/com/mitchelltford/discordbot/GatewayDiscordClientConfig.java:14]: My bad, I mistyped this in the issue
```suggestion
  public GatewayDiscordClient gatewayDiscordClientConfig(@Value("${bot.token}") String token) {
```

**Review by MitchellTFord — APPROVED** [review]

**Pull request merged**